dlc/src/test/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelfTest.scala

```diff
@@ -93,6 +93,76 @@ class BinaryOutcomeDLCWithSelfTest extends BitcoinSAsyncTest {
     }
   }
 
+  def closeTogether(num1: Double, num2: Double): Boolean = {
+    Math.abs(num1 - num2) < 1.0e-6
+  }
+
+  it should "correctly subtract fees proportionally amongst outputs" in {
+    // Can't use TransactionGenerators.realisiticOutputs as that can return List.empty
+    val nonEmptyRealisticOutputsGen = Gen
```

Comment:
```scala
    val nonEmptyRealisticOutputsGen = TransactionGenerators.realisticOutputs
      .suchThat(_.nonEmpty)
```

Reply:
Better not to have a 20% failure with a `suchThat`